apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs

```diff
@@ -103,14 +103,31 @@ internal async Task<T> ExecuteScalarAsync<T>(CancellationToken cancellationToken
 
             internal async Task<SpannerDataReader> ExecuteReaderAsync(CommandBehavior behavior, TimestampBound singleUseReadSettings, CancellationToken cancellationToken)
             {
+                if (CommandTextBuilder.SpannerCommandType != SpannerCommandType.Select
+                    && CommandTextBuilder.SpannerCommandType != SpannerCommandType.Read
+                    && CommandTextBuilder.SpannerCommandType != SpannerCommandType.Dml)
+                {
+                    throw new InvalidOperationException("ExecuteReader functionality is only available for queries, reads and DML commands.");
+                }
+
                 ValidateConnectionAndCommandTextBuilder();
-                ValidateCommandBehavior(behavior);
 
-                if (CommandTextBuilder.SpannerCommandType != SpannerCommandType.Select && CommandTextBuilder.SpannerCommandType != SpannerCommandType.Read)
+                if (CommandTextBuilder.SpannerCommandType == SpannerCommandType.Dml)
                 {
-                    throw new InvalidOperationException("ExecuteReader functionality is only available for queries and reads.");
+                    if (singleUseReadSettings != null)
+                    {
+                        throw new NotSupportedException("singleUseReadSettings cannot be used with DML command.");
+                    }
+
+                    if (behavior != CommandBehavior.Default)
```

Comment:
I don't see why we cannot support CommandBehaviour.CloseConnection here. The connection will be closed when the reader is closed, and that seems plausible here. What am I missing?